### PR TITLE
Dex Modification Modes

### PR DESCRIPTION
**Pull request by xihan123**
This pull request introduces a new feature to handle different dex modification modes in the patching process, along with some minor updates and refactoring. The most important changes include adding support for dex modification modes in the `Patcher` and `LSPatch` classes, updating the UI to allow users to select the dex modification mode, and adding corresponding string resources.

### New Feature: Dex Modification Modes

* Added a new parameter `dexMod` to the `Patcher.Options` class and updated the `Patcher` methods to handle this parameter (`manager/src/main/java/org/lsposed/lspatch/Patcher.kt`). [[1]](diffhunk://#diff-736428f98c9a4121bc8224eacf7011024cbae654a7c0de405dfdfd2ba4555693R18) [[2]](diffhunk://#diff-736428f98c9a4121bc8224eacf7011024cbae654a7c0de405dfdfd2ba4555693R35)
* Introduced a new method `dexModLvStr` to map dex modification levels to user-friendly strings (`manager/src/main/java/org/lsposed/lspatch/ui/page/NewPatchScreen.kt`).
* Updated the `PatchOptionsBody` composable to include a dropdown for selecting the dex modification mode (`manager/src/main/java/org/lsposed/lspatch/ui/page/NewPatchScreen.kt`).
* Added a new property `dexMod` to the `NewPatchViewModel` class and updated the patch submission logic to include this property (`manager/src/main/java/org/lsposed/lspatch/ui/viewmodel/NewPatchViewModel.kt`). [[1]](diffhunk://#diff-e76e2ba5e503f52a793262af7d27a2c47a9ccddbe866db3e2fa267ee3b97726bR42) [[2]](diffhunk://#diff-e76e2ba5e503f52a793262af7d27a2c47a9ccddbe866db3e2fa267ee3b97726bR95)

### Updates to `LSPatch` Class

* Added a new parameter `dexMode` to the `LSPatch` class and updated the patching logic to handle different dex modes (`patch/src/main/java/org/lsposed/patch/LSPatch.java`). [[1]](diffhunk://#diff-93aec550d92d219a811c988bd994763bf99e55cf322501240f2a5f565b1371e9L49) [[2]](diffhunk://#diff-93aec550d92d219a811c988bd994763bf99e55cf322501240f2a5f565b1371e9R81-R83) [[3]](diffhunk://#diff-93aec550d92d219a811c988bd994763bf99e55cf322501240f2a5f565b1371e9R271-R279) [[4]](diffhunk://#diff-93aec550d92d219a811c988bd994763bf99e55cf322501240f2a5f565b1371e9R316)

### String Resources

* Added new string resources for dex modification modes in English, Simplified Chinese, and Traditional Chinese (`manager/src/main/res/values/strings.xml`, `manager/src/main/res/values-zh-rCN/strings.xml`, `manager/src/main/res/values-zh-rTW/strings.xml`). [[1]](diffhunk://#diff-deb054e99ce2325f6ce6f0f13cc9442857f485d879cf1c8ff0baee57f67ac848R89-R91) [[2]](diffhunk://#diff-cb0f7c95c2ad4ce2507b85fb585d913f517e1b3c6ce9b213d8b65ac69c7f44b0R89-R91) [[3]](diffhunk://#diff-0ea3ed762b99116c58d9c17c5fe4e00e59afc93d7efe390ec3343d63b0389c27R93-R95)

### CODE REVIEW

**Comment by JingMatrix** [comment]
Thanks for you work!
I think it is better to use a check box, and call it `Inject loader dex`, with default option off.
For description in the manage, I would like to use
```
For applications with isolated services, such as the render engines of browsers, please turn on option to ensure they work properly.
```
Its Chinese translation could be
```
对那些需要孤立服务进程的应用程序，譬如说浏览器的渲染引擎，请勾选此选项以确保他们正常运行。
```

For command line description, use
```
Inject directly the loder dex file into the original application package
```

**Comment by xihan123** [comment]
QQ20241111-001305

[release.zip](https://github.com/user-attachments/files/17692116/release.zip)

**Comment by JingMatrix** [comment]
Sorry, I am very busy with my thesis recently.
I would like to use a boolean valued varaible to indicate the dex injection option in the Java codes, intead of `0` or `1`, since there will be no other alternative patching approches under consideration.
If you are not available for this modification, I will push changes to your branch in few days (when I find more free time).

Thanks again for you help, I appreciate it!

File: manager/src/main/java/org/lsposed/lspatch/Patcher.kt [hunk 1]
```diff
@@ -11,10 +11,12 @@ import org.lsposed.lspatch.share.PatchConfig
 import org.lsposed.patch.LSPatch
 import org.lsposed.patch.util.Logger
 import java.io.IOException
+import java.util.Collections.addAll
 
 object Patcher {
 
     class Options(
+        private val dexMod: Boolean,
```
Comment (JingMatrix) [inline manager/src/main/java/org/lsposed/lspatch/Patcher.kt:19]: Change `dexMod` to `injectDex`

File: manager/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -90,4 +90,6 @@
     <string name="settings_keystore_wrong_alias">Wrong alias name</string>
     <string name="settings_keystore_wrong_alias_password">Wrong alias password</string>
     <string name="settings_detail_patch_logs">Detail patch logs</string>
+    <string name="patch_dex_mod">Inject loader dex</string>
+    <string name="patch_dex_desc">For applications with isolated services, such as the render engines of browsers, please turn on option to ensure that they work properly.</string>
```
Comment (JingMatrix) [inline manager/src/main/res/values/strings.xml:94]: `please turn on option` -> `please turn on this option`

File: patch/src/main/java/org/lsposed/patch/LSPatch.java [hunk 3]
```diff
@@ -79,6 +78,9 @@ public PatchError(String message, Throwable cause) {
     @Parameter(names = {"-l", "--sigbypasslv"}, description = "Signature bypass level. 0 (disable), 1 (pm), 2 (pm+openat). default 0")
     private int sigbypassLevel = 0;
 
+    @Parameter(names = {"-dex", "--dexmod"}, description = "Inject directly the loder dex file into the original application package")
```
Comment (JingMatrix) [inline patch/src/main/java/org/lsposed/patch/LSPatch.java:81]: For argument option name, it is better to use only `{"--injectdex"}`

File: patch/src/main/java/org/lsposed/patch/LSPatch.java [hunk 4]
```diff
@@ -239,7 +241,8 @@ public void patch(File srcApkFile, File outputFile) throws PatchError, IOExcepti
                 for (StoredEntry entry : srcZFile.entries()) {
                     String name = entry.getCentralDirectoryHeader().getName();
                     if (dstZFile.get(name) != null) continue;
-                    if (name.startsWith("META-INF") && (name.endsWith(".SF") || name.endsWith(".MF") || name.endsWith(".RSA"))) continue;
+                    if (name.startsWith("META-INF") && (name.endsWith(".SF") || name.endsWith(".MF") || name.endsWith(".RSA")))
+                        continue;
```
Comment (JingMatrix) [inline patch/src/main/java/org/lsposed/patch/LSPatch.java:245]: Why a newline here?

File: patch/src/main/java/org/lsposed/patch/LSPatch.java [hunk 5]
```diff
@@ -266,11 +269,15 @@ public void patch(File srcApkFile, File outputFile) throws PatchError, IOExcepti
 
             logger.i("Adding metaloader dex...");
             try (var is = getClass().getClassLoader().getResourceAsStream(Constants.META_LOADER_DEX_ASSET_PATH)) {
-                var dexCount = srcZFile.entries().stream().filter(entry -> {
-                    var name = entry.getCentralDirectoryHeader().getName();
-                    return name.startsWith("classes") && name.endsWith(".dex");
-                }).collect(Collectors.toList()).size() + 1;
-                dstZFile.add("classes" + dexCount + ".dex", is);
+                if (!dexMode) {
```
Comment (JingMatrix) [inline patch/src/main/java/org/lsposed/patch/LSPatch.java:272]: Change the variable `dexMode`

**Review by JingMatrix — CHANGES_REQUESTED** [review]
You may need to change few variable names to keep consistent and easy to understand for others.

File: patch/src/main/java/org/lsposed/patch/LSPatch.java (hunk 4 above)
Comment (xihan123) [reply patch/src/main/java/org/lsposed/patch/LSPatch.java:245]: IDEA code optimization

**Comment by JingMatrix** [comment]
Congratulations to your contributions, it is merge! Thanks for your patience, I appreciate it!

**Pull request merged**